vault_service/views/user.py

```diff
@@ -160,9 +160,21 @@ def store_data():
                 return '{}', 200 # or return 404?
             return q.user_data or '{}', 200
     elif request.method == 'POST':
-        d = json.dumps(payload)
+        # check if user record exists first
+        try:
+            with current_app.session_scope() as session:
+                q = session.query(User).filter_by(id=user_id).first()
```

Comment:
If between this reading and the writing there is another request that changes the data in the DB, we will lose it. After a quick googling, it seems something like this is allowed in sqlalchemy: `session.query(User).filter_by(id=user_id).first().update({"values": values})`. This will make us resilient to race conditions, could you explore if this or something similar can work?

Reply:
The .update() just overwrites (and doesn't merge) the given field, so won't work here. The later exception on the Integrity Error provides some protection, but I'll see what else can be done for race conditions.